src/app.js

```diff
@@ -22,17 +26,31 @@ export function app(app) {
     })
   }
 
-  emit("ready", render(state, view))
+  render()
+
+  emit("init")
 
   return emit
 
-  function render(state, view) {
-    element = patch(
-      root,
-      element,
-      node,
-      (node = emit("render", view)(state, actions))
-    )
+  function render() {
+    if (!locked) {
+      locked = !locked
+
+      requestAnimationFrame(function() {
+        element = patch(
+          root,
+          element,
+          node,
+          (node = emit("render", view)(state, actions))
+        )
+
+        locked = !locked
```

Comment:
I've just realized we may be thinking about this wrong. 

Anything which needs to happen before the first render should be in the "init" event.

It's probably a good thing that the "loaded" event causes another render, otherwise those 2 events would be identical.

In conclusion.... nevermind! :laughing: